airflow/www/views.py

```diff
@@ -2886,21 +2886,22 @@ def duration(self, dag_id, session=None):
             min_date = timezone.utc_epoch()
         ti_fails = (
             session.query(TaskFail)
+            .join(TaskFail.dag_run)
             .filter(
                 TaskFail.dag_id == dag.dag_id,
                 DagRun.execution_date >= min_date,
                 DagRun.execution_date <= base_date,
-                TaskFail.task_id.in_([t.task_id for t in dag.tasks]),
             )
-            .all()
         )
+        if dag.partial:
+            ti_fails = ti_fails.filter(TaskFail.task_id.in_([t.task_id for t in dag.tasks]))
```

Comment:
This one doesn’t seem related?

Reply:
Not related to the fix/exception, but it was missing from gantt but done in durations.

(Yes, there is too much duplication between those views.)

Reply:
Wait -- no sorry, mis-remembering.

This is a drive-by-optimization.

There's no point specfying all the task ids if we're just going to include everything. So I now only do it if we're not using the full dag.